internal/tofu/context_plan2_test.go

```diff
@@ -4533,6 +4533,191 @@ import {
 	}
 }
 
+func TestContext2Plan_importForEach(t *testing.T) {
```

Comment:
I didn't include "unhappy" tests like having a null key/value, sensitive key/value, or unknown key/value, as they fall under existing tests in either `eval_for_each_test.go` `TestEvaluateForEachExpression_errors()` and `context_plan2_test.go` `TestContext2Plan_importToInvalidDynamicAddress()`.
I tested them all manually with all possible for_each types (`map`, `object`, `set`, `tuple`), but didn't want to bloat our existing test cases with duplicated cases. Let me know if you still think we should add these tests for the for_each scenario.

Reply:
I think we should add them, mainly because it would help us identify regressions in the future, especially if eval_for_each_test.go somehow gets refactored. I would rather have some duplication of tests if it gives us faith that we're testing properly.

Reply:
Though I think this doesn't block the PR from the beta, and we can possibly add it in a follow-up PR (if it's a lot of work; if it's not let's just do it).

Reply:
It shouldn't take long, but I'll push those tests as part of a different PR since we want this one merged by tomorrow.